### PR TITLE
appveyor: build test suite for coff32

### PR DESCRIPTION
**Pull request by rainers**

### CODE REVIEW

**Comment by wilzbach** [comment]
Looks like it's definitely worthwhile to do this:

```
Test runnable/test17559.d failed.  The logged output:
..\generated\Windows\Release\Win32\dmd.exe -conf= -m32mscoff -Irunnable -g    -odtest_m32mscoff\runnable -oftest_m32mscoff\runnable\test17559_0.exe  runnable\test17559.d  -map nul.map
test_m32mscoff\runnable\test17559_0.exe
object.Exception@runnable\test17559.d(48): fun
----------------
0x003DADEE in test17559 at C:\projects\dmd\test\runnable\test17559.d(48)
0x003DAD3B in D at C:\projects\dmd\test\runnable\test17559.d(32)
0x003DBE67 in void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll().__lambda1()
0x003DBDA9 in void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll()
0x003DBB68 in _d_run_main
0x003DB2BC in __entrypoint
0x0040CC8B in __scrt_common_main_seh at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl(253)
0x76E77C04 in BaseThreadInitThunk
0x77D5AD8F in RtlInitializeExceptionChain
0x77D5AD5A in RtlInitializeExceptionChain
core.exception.AssertError@runnable\test17559.d(55): Assertion failure
----------------
0x003DB463 in _d_assertp
0x003DAD3B in D at C:\projects\dmd\test\runnable\test17559.d(32)
0x003DBE67 in void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll().__lambda1()
0x003DBDA9 in void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll()
0x003DBB68 in _d_run_main
0x003DB2BC in __entrypoint
0x0040CC8B in __scrt_common_main_seh at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl(253)
0x76E77C04 in BaseThreadInitThunk
0x77D5AD8F in RtlInitializeExceptionChain
0x77D5AD5A in RtlInitializeExceptionChain
```

**Comment by skl131313** [comment]
Does the VisualD build for win32 use `-m32` or `-m32mscoff`?

**Comment by rainers** [comment]
> Does the VisualD build for win32 use -m32 or -m32mscoff?

It links with a VC built backend, so it has to use -m32mscoff.

**Review by JinShil — APPROVED** [review]

File: test/runnable/test15779.d [hunk 1]
```diff
@@ -18,5 +18,6 @@ int main()
 
 void bar()
 {
+    version(CRuntime_Microsoft) {} else // TODO: fails for -O -inline, also disabled for the auto-tester
```
Comment (JinShil) [inline test/runnable/test15779.d]: What's the problem here?  What's the plan for eventually enabling it for the auto tester?  Is a bug report filed for it?

Comment (rainers) [reply test/runnable/test15779.d]: Not sure, maybe related: https://issues.dlang.org/show_bug.cgi?id=13821 and https://issues.dlang.org/show_bug.cgi?id=16511.
I have added a new one: https://issues.dlang.org/show_bug.cgi?id=18547

Comment (JinShil) [reply test/runnable/test15779.d]: In that case I recommend adding links to those issues in the comments and come clean about you're understanding of what's going on.  Something like:

"Fibers appear to be broken in Windows.  We don't yet have a solution, see these issues...."  or something like that.  My motivations is to ensure that there is some kind of "paper trail" that one can follow to eventually solve the issue.

**Comment by skl131313** [comment]
Using `DISABLED: win32` in the test settings at the top would help for that a bit, so that the test output doesn't show it's successful rather than being skipped.

File: test/runnable/test15779.d (hunk 1 above)
Comment (WalterBright) [reply test/runnable/test15779.d]: I agree with the "paper trail" proposal, otherwise such comments just mystify the reader.

Comment (ZombineDev) [reply test/runnable/test15779.d]: @rainers my guess is that this test passes on LDC. Can you scope this to CRuntime_Microsoft and DigitalMars (i.e. dmd)?

Comment (rainers) [reply test/runnable/test15779.d]: I think I figured it out: https://github.com/dlang/druntime/pull/2129. All windows builds should be affected, but happens spuriously.

Comment (rainers) [reply test/runnable/test15779.d]: I have removed the workaround in the hope that the bugfix will be accepted.

Comment (ZombineDev) [reply test/runnable/test15779.d]: Thanks!

**Comment by rainers** [comment]
Appveyor even passed without https://github.com/dlang/druntime/pull/2129, but I guess that can change spuriously (but also for the Win64 check on Appveyor or the Windows checks on the auto tester).

**Comment by JinShil** [comment]
Restarted Semaphore CI.  All CIs are passing now.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**